VueJS/index.js

```diff
@@ -1,13 +1,72 @@
-import List from './components/List.js'
-import ListItem from './components/ListItem.js'
+// import List from './components/List.js'
+// import ListItem from './components/ListItem.js'
 
-new Vue({
-  el: '#app',
-  data: {
-    todos: [
-      { id: '1', message: 'Foo', },
-      { id: '2', message: 'Bar', },
-      { id: '3', message: 'Baz', }
-    ]
+Vue.component('list-item', {
+  props: ['todo'],
+  template: `
+      <div v-if="edit">
+        <li>
+          <input id="input" v-model="newTodo">
+          <button v-on:click="saveTodo">Save</button>
+          <button v-on:click="cancelTodo">Cancel</button>
+          <button v-on:click="$emit('deleteTodo')">Delete</button>
+        </li>
+      </div>
+      <div v-else>
+        <li>
+        {{ todo.message }}
+        <button v-on:click="editTodo">Edit</button>
+        <button v-on:click="$emit('deleteTodo')">Delete</button>
+        </li>
+      </div>
+  `,
+  data: function(){
+    return{
+      edit: false,
+      newTodo: ''
+    }
+  },
+  methods: {
+    editTodo: function(){
+      this.edit = true
+    },
+    cancelTodo: function(){
+      this.edit = false
+    },
+    saveTodo: function(){
+      this.todo.message = document.getElementById("input").value;
```

Comment:
Avoid globals like `document` and `window`. Those are not available in server-side rendering, ie. outside the browser environment. Also it is quite an anti-pattern to manipulate the browser DOM when you have a virtual DOM with VueJS.

![yuck](https://media.giphy.com/media/ulJqb38U1W0jhU80sD/giphy.gif)

Read: https://vuejs.org/v2/guide/components.html#Using-v-model-on-Components

Reply:
Somehwere up there :point_up: you are even using `v-model` already with `newTodo`. Have you checked the value of `newTodo` here? You could emit that object to the parent list, so it can then update the list of data objects which will in turn trigger a re-rendering of all `<list-item>` components.